src/core/thread/thread_netif.cpp

```diff
@@ -49,7 +49,7 @@ void ThreadNetif::Up(void)
 
     // Enable the MAC just in case it was disabled while the Interface was down.
     Get<Mac::Mac>().SetEnabled(true);
-#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE && OPENTHREAD_CONFIG_CHANNEL_MONITOR_ON_STARTUP_ENABLE
```

Comment:
Should we add the same check on `Stop()` as well?

Reply:
Do any issues arise if the channel monitor is left running after the interface is brought down?

Reply:
`ChannelMonitor` uses the `Mac` directly and invokes `EnergyScan()` to get RSS samples, so I don't expect any issues. That said, I would recommend testing/validating this (to be safe).

Reply:
We've been running a [variant](https://github.com/home-assistant/addons/blob/master/openthread_border_router/0001-channel-monitor-disable-by-default.patch) of this patch for a little over a year now without any noticeable problems so I'll take a look at the test failures and see if this feature can be tested explicitly as well.

Reply:
Hm, so that would mean that energy scan would continue to run when the Thread interface is stopped? It seems almost this is a different use case.

If we go that route, maybe the flag should not only mention startup enable, but be more about automatically control monitor start/stop, e.g. `OPENTHREAD_CONFIG_CHANNEL_MONITOR_AUTOSTART` maybe?

Reply:
Yes, I proposed the following:

```
 * If enabled, the Channel Monitor will start and stop automatically when the Thread network interface is brought up
 * and down.
 ```

We already start it in `ThreadNetif::Up()` and stop it in `ThreadNetif::Down()` in the existing code. The idea is that the new `OPENTHREAD_CONFIG_CHANNEL_MONITOR_AUTO_START_ENABLE` configuration option would make this behavior configurable at build time, thus leaving it up to the next layer of code to explicitly start and stop the channel monitor using the public API when it wants.